AT3/AT3Tags.cpp

```diff
@@ -11,19 +12,50 @@ AT3Tags::AT3Tags() : CPlugIn(EuroScopePlugIn::COMPATIBILITY_CODE, MY_PLUGIN_NAME
 	RegisterTagItemType("AT3 Assigned Altitude", TAG_ITEM_AT3_ALTITUDE_ASSIGNED);
 	RegisterTagItemType("AT3 Track", TAG_ITEM_AT3_TRACK);
 	RegisterTagItemType("AT3 Assigned Heading", TAG_ITEM_AT3_HEADING_ASSIGNED);
-	RegisterTagItemType("AT3 Groundspeed", TAG_ITEM_AT3_SPEED);
+	RegisterTagItemType("AT3 Ground Speed", TAG_ITEM_AT3_SPEED);
 	RegisterTagItemType("AT3 Assigned Speed (IAS)", TAG_ITEM_AT3_SPEED_ASSIGNED);
-	RegisterTagItemType("AT3 TMA Entry Point", TAG_ITEM_AT3_TMA_ENTRY);
-	RegisterTagItemType("AT3 Approach Type", TAG_ITEM_AT3_APP_TYPE);
+	RegisterTagItemType("AT3 Route Code", TAG_ITEM_AT3_ROUTE_CODE);
+	RegisterTagItemType("AT3 APP/DEP Line 4", TAG_ITEM_AT3_APPDEP_LINE4);
+	RegisterTagItemType("AT3 AMC Line 4", TAG_ITEM_AT3_AMC_LINE4);
 	RegisterTagItemType("AT3 Slot", TAG_ITEM_AT3_SLOT);
+	RegisterTagItemType("AT3 Callsign", TAG_ITEM_AT3_CALLSIGN);
+	RegisterTagItemType("AT3 ATYP + WTC", TAG_ITEM_AT3_ATYPWTC);
+	RegisterTagItemType("AT3 VS Indicator", TAG_ITEM_AT3_VS_INDICATOR);
+	RegisterTagItemType("AT3 Arrival Runway", TAG_ITEM_AT3_ARRIVAL_RWY);
 }
 
+
 void AT3Tags::OnGetTagItem(CFlightPlan FlightPlan, CRadarTarget RadarTarget, int ItemCode, int TagData, char sItemString[16], int* pColorCode, COLORREF* pRGB, double* pFontSize)
 {
 	if (!FlightPlan.IsValid() || !RadarTarget.IsValid()) {
 		return;
 	}
 
+	*pColorCode = TAG_COLOR_RGB_DEFINED;
+	switch (FlightPlan.GetState()) {
+		case FLIGHT_PLAN_STATE_NON_CONCERNED:
+			*pRGB = RGB(117, 132, 142);
+			break;
+		case FLIGHT_PLAN_STATE_NOTIFIED:
+			*pRGB = RGB(117, 132, 142);
+			break;
+		case FLIGHT_PLAN_STATE_COORDINATED:
+			*pRGB = RGB(117, 132, 142);
+			break;
+		case FLIGHT_PLAN_STATE_TRANSFER_TO_ME_INITIATED:
+			*pRGB = RGB(229, 214, 130);
+			break;
+		case FLIGHT_PLAN_STATE_TRANSFER_FROM_ME_INITIATED:
+			*pRGB = RGB(241, 246, 255);
+			break;
+		case FLIGHT_PLAN_STATE_ASSUMED:
+			*pRGB = RGB(241, 246, 255);
+			break;
+		case FLIGHT_PLAN_STATE_REDUNDANT:
+			*pRGB = RGB(229, 214, 130);
```

Comment:
These colors shouldn't be hardcoded in the future, but I think we need to redefine all the colors as member variables anyway so can save for future release